mlflow/server/js/src/components/ExperimentView.js

```diff
@@ -113,6 +126,26 @@ class ExperimentView extends Component {
     return true;
   }
 
+  /** Returns true if search filter text was updated */
+  filtersDidUpdate(prevState) {
+    return prevState.paramKeyFilterInput !== this.state.paramKeyFilterInput ||
```

Comment:
Why are these state instead of props (assuming here that ExperimentPage wraps ExperimentView and the former has these as state fields)?

Reply:
TL;DR: Here we're avoiding snapshotting component state when the user types new text into the metric/param filter boxes (i.e. we don't need to snapshot on each keystroke, only when the user presses "Search").

Detail: yeah this logic is kinda confusing, there's (in ExperimentPagePersistedState):
```
  // Comma-separated string containing containing the keys of parameters to display
  paramKeyFilterString: "",
  // Comma-separated string containing containing the keys of metrics to display
  metricKeyFilterString: "",
```

And in ExperimentView's props:
```
    // Input to the paramKeyFilter field
    paramKeyFilter: PropTypes.instanceOf(KeyFilter).isRequired,
    // Input to the paramKeyFilter field
    metricKeyFilter: PropTypes.instanceOf(KeyFilter).isRequired,
```

And in ExperimentView's unpersisted state (the initial values of these state fields are actually derived from the `metricKeyFilter` and `paramKeyFilter` props):
```
      // Text entered into the param filter field
      paramKeyFilterInput: '',
      // Text entered into the metric filter field
      metricKeyFilterInput: '',
```

`ExperimentPage` maintains the last-submitted (via pressing the "search" button) metric & param filter strings as component state, passing these along to `ExperimentView` via `KeyFilter` props that can be used to filter which columns appear in the table (the filtering logic will eventually happen server-side  / outside ExperimentView, once we add metric/param filters to the search API). ExperimentView also maintains the text entered into the metric/search filter boxes as unpersisted state so it can pass that text to ExperimentPage on search events.